src/main/java/org/jenkinsci/plugins/envinject/service/PropertiesLoader.java

```diff
@@ -68,8 +68,12 @@
     @Nonnull
     private Map<String, String> getVars(@Nonnull String content, @Nonnull Map<String, String> currentEnvVars) 
             throws EnvInjectException {
+
+        // Replace single backslashes with double ones so they won't be removed by Property.load()
+        String escapedContent = content;
+        escapedContent = escapedContent.replaceAll("(?<=[^\\\\])\\\\(?=[^n])(?=[^\\\\])(?=[^\n])", "\\\\\\\\");
```

Comment:
```suggestion
        escapedContent = escapedContent.replaceAll("(?<=[^\\\\])\\\\(?=[^n|\\W])(?=[^\\\\])(?=[^\n])", "\\\\\\\\");
```
If you do this you would not have to change the github example because that will cause a backwards incompatibility

Reply:
The negative lookahead will fix trailing backslash

Reply:
Actually looking at it again it might cause issues so i reverted that bit